server/cluster/cluster.go

```diff
@@ -588,6 +589,8 @@ func (c *RaftCluster) HandleStoreHeartbeat(stats *pdpb.StoreStats) error {
 	}
 	// Here we will compare the reported regions with the previous hot peers to decide if it is still hot.
 	c.hotStat.CheckReadAsync(statistics.NewCollectUnReportedPeerTask(storeID, regions, interval))
+
+	c.handleDamagedStore(stats)
```

Comment:
It seems that this function doesn't need to acquire the lock, right? I don't think putting it here is a good idea.

Reply:
commented below